src/test/java/io/jenkins/plugins/analysis/core/steps/IssuesRecorderTest.java

```diff
@@ -54,4 +64,300 @@ void shouldContainSingleElementAndPlaceHolder() {
         assertThat(actualModel).hasSize(2);
         assertThat(actualModel).containsExactly("-", name);
     }
+
+
+    /**
+     * Verifies that the validation is ok if the specified source code encoding is a valid encoding
+     */
+    @Test
+    void shouldBeOkForValidCharsets(){
+        final Descriptor descriptor = new Descriptor();
+
+        final FormValidation actualResult = descriptor.doCheckSourceCodeEncoding(validEncoding);
+
+        assertThat(actualResult).isOk();
+    }
+
+    /**
+     * Verifies that the validation is marked as an error if the specified source code encoding is invalid
+     */
+    @Test
+    void shouldNotBeOkIfSpecifiedCharsetIsInvalid(){
+        final String inValidEncoding = "encodingThatWillNeverExits";
+        final Descriptor descriptor = new Descriptor();
+
+        final FormValidation actualResult = descriptor.doCheckSourceCodeEncoding(inValidEncoding);
+
+        assertThat(actualResult).isError();
+        assertThat(actualResult).hasMessage("Encoding must be a supported encoding of the Java platform, see: <a href=\"https://docs.oracle.com/javase/8/docs/api/java/nio/charset/Charset.html\">java.nio.charset.Charset</a>");
+    }
+
+    /**
+     * Verifies that the getDisplayName is returning the right name
+     */
+    @Test
+    void shouldReturnRightName(){
+        final Descriptor descriptor = new Descriptor();
+
+        final String actualName = descriptor.getDisplayName();
+
+        assertThat(actualName).isEqualTo(Messages.ScanAndPublishIssues_DisplayName());
```

Comment:
Nicht falsch aber kann in Zukunft weggelassen werden, da Duplizieren der tatsächlichen Implementierung.